### DISCUSSION TITLE
Welcome to instascrape Discussions!

### DISCUSSION THREAD

**Discussion by chris-greening** [chris-greening/instascrape#47] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

# whoami
Heyo everyone! My name is Chris Greening and I'm the author/maintainer of `instascrape`. I am from Long Island, New York and have a BSc in Physics. I began my programming journey in 2017 after taking an introduction to computational physics course and have since spent all my free time learning as much as I can about software development. When I'm not programming, I am incredibly passionate about [photography](https://www.instagram.com/chris_greening/), video games, roller blading, reading, hiking, fishing, and a whole bunch of other hobbies that take up just about 100% of my time lol 

# A little bit about instascrape 
`instascrape` started in Summer 2019 as a small, poorly organized script that scraped the upload datetime and amount of likes every post on an individual profile had. I posted about it on my Instagram story and had a bunch of random people asking me to show them their profile engagement and it made me realize that this was a tool people wanted. 

Fast forward to Hacktoberfest 2020; despite the backlash the event had, I was excited to get involved in open source and so I started working on releasing my scraper to the general public. I quickly realized that the original code was awful so I reevaluated it and started entirely from scratch. It has been a heck of a ride and countless hours over the last few months getting to the point we're at now but it is a library I'm proud of and I hope it's something the community can be proud of too. 

The goal of the library is to be expressive and concise. This is a flexible set of tools designed to fit a lot of different use cases so I hope it can serve you well... and if it doesn't? Then let me know what I can do to make it better! 

#  hbu
I absolutely love talking and interacting with others so feel free to introduce yourself in the comments here or in a separate discussion! Ideas, comments, fixes, features, concerns, etc. are all welcome. 

Regards,
Chris Greening

**Comment by gabrielarcangelbol** [comment]
Thanks for create this space to talk about  Instascrape. I´m learning about programming in python. And i´m doing a project to make some data analysis using Instagram as a social network, i found Instascrape easy to use. I would like to use Instascrape to visualize instagram engagement and do sentiment analysis as well. Thanks again!

**Comment by chris-greening** [comment]
Hey @gabrielarcangelbol, can't wait to see what you come up with! I use instascrape a lot for data analysis myself lol and it's actually one of the primary inspirations behind the lib so hopefully you'll get some good use out of it, let me know if you have any Python or instascrape questions 😎

**Comment by tarob0ba** [comment]
Hello! I am currently working on improving my Python skills and contributing to open source software. My first ever contribution to open source was to _instascrape_ and it helped me feel confident contributing elsewhere. _Instascrape_ is a valuable tool for a data scientist and I can't wait to see where it goes. On a side note, I'm currently working on a tiny programming language called Creeper, named for my love of a classic videogame, Minecraft. 😎

**Reply by chris-greening** [reply]
Hey Beckett! I'm so glad that you got to use `instascrape` as a means for getting comfortable with open source; this project has helped me feel more comfortable with it as well 😄 You were one of if not the first contributors to the project and it encouraged me greatly knowing at least someone out there was tinkering around with my code, thanks so much for that! 

What are some of your goals for Creeper?

**Reply by tarob0ba** [reply]
My major goals for Creeper are as follows:

- Code written in Creeper is easy to read, write, and comprehend
- Code written in Creeper is dynamic
- Code written in Creeper is small, but without losing readability.

Essentially, I want Creeper to follow some of Python's ideals, but to give flexibility in size of language **and** in size of program. From what I have seen, it is difficult or impossible to minify Python. (If for some reason you forgot about the Zen of Python, were running out of storage, or were trying to make code generated by Brython or Transcrypt smaller before minifying it further.)

Finally, my ultimate goal for Creeper is for the language to be fun and enjoyable to read. By incorporating readability with logic and size, I want Creeper to be good for writing small but powerful programs. (Obviously, Creeper is nowhere near that, but it's nice to dream.)

**Reply by chris-greening** [reply]
Awesome! Aside from loosely digging through CPython's source, my experience with programming language design and development is really limited. I love learning new things though so I'm excited to see where you go with this project! I'll keep an eye on it 👁️
